src/pmemkv.h

```diff
@@ -56,9 +56,18 @@ const string LAYOUT = "pmemkv";                            // pool layout identi
 
 class KVEngine {                                           // storage engine implementations
   public:
+    struct Options {
+        bool create_if_missing = false;                    // create persistent pool if it is missing
+        bool error_if_exists = false;                      // raise error if persistent pool exists
```

Comment:
Those names are a bit confusing...

Maybe change it to a bitmask of "Open" and "Create"?
Open (= pmemobj_open)
Create (= pmemobj_create)
Open | Create (pmemobj_open, on failure pmemobj_create)

Using a struct doesn't buy us much, because you won't be able to add new fields (ABI changes).

Reply:
So:
- false / false = Open
- true / true = Create
- false / true = Open | Create
- true / false does not have an equivalent. It means "always throw an error, independent of whether pool exists or not". Not very useful behavior ;)

If compatibility with RocksDB WRT options naming is really important, then I guess create_if_missing/error_if_exists is good enough. Whichever option Rob will choose is going to be better than what we have on master.

Reply:
Are we looking at the same code on master?
https://github.com/pmem/pmemkv/blob/master/src/engines/kvtree2.cc#L47

I think maybe you were looking at the old kvtree engine from 2017?